pkg/trace/api/evp_intake_proxy_test.go

```diff
@@ -0,0 +1,203 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type roundTripperMock func(*http.Request) (*http.Response, error)
+
+func (r roundTripperMock) RoundTrip(req *http.Request) (*http.Response, error) {
+	return r(req)
+}
+
+// sendRequestThroughHandler sends a request through the evpIntakeReverseProxy handler and returns the forwarded
+// request(s), their response and the log output. The path for inReq shouldn't have the /evpIntakeProxy/v1 prefix
+// since it is passed directly to the inner proxy handler and not the trace-agent API handler.
+func sendRequestThroughHandler(conf *config.AgentConfig, inReq *http.Request) (outReqs []*http.Request, response *http.Response, loggerOut string) {
+	mockRoundTripper := roundTripperMock(func(req *http.Request) (*http.Response, error) {
+		outReqs = append(outReqs, req)
+		// If we got here it means the proxy didn't raise an error earlier, return an ok response
+		return &http.Response{
+			StatusCode: 200,
+			Body:       ioutil.NopCloser(bytes.NewBuffer([]byte("ok_responserino"))),
+		}, nil
+	})
+	var loggerBuffer bytes.Buffer
+	mockLogger := log.New(io.Writer(&loggerBuffer), "", 0)
+	handler := evpIntakeReverseProxyHandler(conf, evpIntakeEndpointsFromConfig(conf), mockRoundTripper, mockLogger)
+	rec := httptest.NewRecorder()
+	handler.ServeHTTP(rec, inReq)
+	return outReqs, rec.Result(), loggerBuffer.String()
+}
+
+func TestEvpIntakeReverseProxyHandlerOk(t *testing.T) {
+	conf := newTestReceiverConfig()
+	conf.Hostname = "test_hostname"
+	conf.DefaultEnv = "test_env"
+	conf.Site = "us3.datadoghq.com"
+	conf.Endpoints[0].APIKey = "test_api_key"
+
+	request := httptest.NewRequest("POST", "/mysubdomain/mypath/mysubpath?arg=test", nil)
+	request.Header.Set("User-Agent", "test_user_agent")
+	request.Header.Set("Content-Type", "text/json")
+	request.Header.Set("Unexpected-Header", "To-Be-Discarded")
+	proxiedRequests, response, loggerOut := sendRequestThroughHandler(conf, request)
```

Comment:
Same here, let's shorten them a bit. Proposing
```suggestion
	proxyreq, resp, logs := sendRequestThroughHandler(conf, request)
```